fabric_generator/fabulous_top_wrapper_temp/top_wrapper_generator_with_BRAM.py

```diff
@@ -58,356 +46,551 @@ def main(argv):
+        BRAM_str += '\t.rd_addr(FAB2RAM_A[' + \
+            str(addr_cap*i+8-1)+':'+str(addr_cap*i)+']),\n'
+        BRAM_str += '\t.rd_data(RAM2FAB_D[' + \
+            str(data_cap*i+32-1)+':'+str(data_cap*i)+']),\n'
+        BRAM_str += '\t.wr_addr(FAB2RAM_A['+str(addr_cap *
+                                                i+16-1)+':'+str(addr_cap*i+8)+']),\n'
+        BRAM_str += '\t.wr_data(FAB2RAM_D[' + \
+            str(data_cap*i+32-1)+':'+str(data_cap*i)+']),\n'
+        BRAM_str += '\t.C0(FAB2RAM_C['+str(config_cap*i)+']),\n'
+        BRAM_str += '\t.C1(FAB2RAM_C['+str(config_cap*i+1)+']),\n'
+        BRAM_str += '\t.C2(FAB2RAM_C['+str(config_cap*i+2)+']),\n'
+        BRAM_str += '\t.C3(FAB2RAM_C['+str(config_cap*i+3)+']),\n'
+        BRAM_str += '\t.C4(FAB2RAM_C['+str(config_cap*i+4)+']),\n'
+        BRAM_str += '\t.C5(FAB2RAM_C['+str(config_cap*i+5)+'])\n'
+        BRAM_str += '\t)\n\n'
+
+    wrapper_top_str += BRAM_str
+
+    wrapper_top_str += "\tassign FrameData = {32'h12345678,FrameRegister,32'h12345678}\n\n"
+    wrapper_top_str += 'endmodule\n\n'
 
     # wrapper_top_str+='module sky130_fd_sc_hd__inv (\n'
     # wrapper_top_str+='\tY,\n'
     # wrapper_top_str+='\tA\n'
-    # wrapper_top_str+='\t);\n'
-    # wrapper_top_str+='\toutput Y;\n'
-    # wrapper_top_str+='\tinput  A;\n'
+    # wrapper_top_str+='\t)\n'
+    # wrapper_top_str+='\toutput Y\n'
+    # wrapper_top_str+='\tinput  A\n'
     # wrapper_top_str+='\n'
-    # wrapper_top_str+='\tassign Y=~A;\n'
+    # wrapper_top_str+='\tassign Y=~A\n'
     # wrapper_top_str+='endmodule\n'
-    #wrapper_top_str+=data_reg_modules
-    #wrapper_top_str+=strobe_reg_modules
-        
+    # wrapper_top_str+=data_reg_modules
+    # wrapper_top_str+=strobe_reg_modules
+
     if wrapper_top_str:
-        with open("eFPGA_top.v", 'w') as file:
+        with open(f"{output_dir}/eFPGA_top.v", 'w') as file:
             file.write(wrapper_top_str)
 
     if data_reg_modules:
-        with open("Frame_Data_Reg_Pack.v", 'w') as file:
+        with open(f"{output_dir}/Frame_Data_Reg_Pack.v", 'w') as file:
             file.write(data_reg_modules)
 
     if strobe_reg_modules:
-        with open("Frame_Select_Pack.v", 'w') as file:
+        with open(f"{output_dir}/Frame_Select_Pack.v", 'w') as file:
             file.write(strobe_reg_modules)
-        
+
     if config_str:
-        with open("Config.v", 'w') as file:
+        with open(f"{output_dir}/Config.v", 'w') as file:
             file.write(config_str)
-        
+
     if configfsm_str:
-        with open("ConfigFSM.v", 'w') as file:
+        with open(f"{output_dir}/ConfigFSM.v", 'w') as file:
             file.write(configfsm_str)
-        
-    #if testbench_str:
+
+    # if testbench_str:
     #    with open("tb_bitbang.vhd", 'w') as file:
     #        file.write(testbench_str)
-    
+
     print("Finish")
 
+
 if __name__ == "__main__":
-    main(sys.argv[1:])
 
+    if sys.version_info <= (3, 5, 0):
```

Comment:
as there seem to be some f-strings already above, this should actually be 3.6 not 3.5 I think